app/controllers/income/previousYears/UpdateIncomeDetailsController.scala

```diff
@@ -109,7 +109,7 @@ class UpdateIncomeDetailsController @Inject()(
       },
       incomeDetails => {
         journeyCacheService
-          .cache(Map(UpdatePreviousYearsIncome_IncomeDetailsKey -> incomeDetails))
+          .cache(Map(UpdatePreviousYearsIncome_IncomeDetailsKey -> incomeDetails.replace("\r", "")))
```

Comment:
Do we need this bit? The validation should have stripped it by the time it comes to caching?

Reply:
I checked and it doesn't, after the validation there's still the additional character present when we come to the controller